INDI/INDI/indiserver.c

```diff
@@ -838,46 +824,57 @@ static void shutdownDvr(DvrInfo *dp, int restart)
         /* close local named FIFOs */
         close(dp->wfd);
         close(dp->rfd);
-#       if 0
-        close(dp->efd);
-#       endif/*0*/
     }
 
 #ifdef OSX_EMBEDED_MODE
     fprintf(stderr, "STOPPED \"%s\"\n", dp->name);
     fflush(stderr);
 #endif
 
-    /* free memory */
-    free(dp->sprops);
-    free(dp->dev);
-    delLilXML(dp->lp);
+    /* free memory; ensure no double-free if stop interrupts restart */
+    if (dp->sprops) { free(dp->sprops); dp->sprops = NULL; }
+    if (dp->dev) { free(dp->dev); dp->dev = NULL; }
+    if (dp->lp) { delLilXML(dp->lp); dp->lp = NULL; }
```

Comment:
shutdownDvr is called on a driver that is waiting to be restarted, and it may be called again before the restart attempt.

This new code ensures re/c/malloc'ed memory is not passed to free more than once.